src/common/telemetry/src/metric.rs

```diff
@@ -25,3 +33,407 @@ pub fn dump_metrics() -> Result<String, String> {
         .map_err(|_| "Encode metrics failed".to_string())?;
     String::from_utf8(buffer).map_err(|e| e.to_string())
 }
+
+/// `MetricFilter` used in `report_metric_task`.
+/// for metric user don't want collect, return a `false`, else return a `true`
+pub type MetricFilter = Box<dyn Fn(&MetricFamily) -> bool>;
+
+pub fn report_metric_task(config: RemoteWriteOptions) {
+    if !config.enable {
+        return;
+    }
+    if config.write_interval.as_secs() == 0 {
+        panic!("Expected Remote write write_interval greater than zero")
+    }
+    let _ = std::thread::spawn(move || loop {
+        std::thread::sleep(config.write_interval);
```

Comment:
Prefer not to start a physical thread on each background task itself. Use `spawn_bg` instead